### PR TITLE
NEW: Add exclude callable to `cached` decorator

### PR DESCRIPTION
**Pull request by JeroennC**

### CODE REVIEW

File: tests/decorators/test_async.py [hunk 1]
```diff
@@ -118,3 +118,34 @@ async def expensive_function(**kwargs: Any) -> int:
...
+
+def sync_exclude(key_, value_):
+    return int(key_) + value_ < 40
+
+
+async def async_exclude(key_, value_):
+    return sync_exclude(key_, value_)
+
+
+@pytest.mark.parametrize(
```
Comment (eigenein) [inline tests/decorators/test_async.py:131]: This looks somewhat excessive since both cases are now covered by `maybe_awaitable`, so you could shorten this to a simple inline lambda for `exclude`. You probably replicated `sync_ttl`/`async_ttl` and I'm going to refactor them anyway

File: tests/decorators/test_async.py [hunk 2]
```diff
@@ -118,3 +118,34 @@ async def expensive_function(**kwargs: Any) -> int:
...
+        exclude=exclude_function,
+    )
+    async def power_function(arg: int) -> int:
+        return arg**2
+
+    with mock.patch.object(cache, "set", wraps=cache.set) as m_set:
+        assert await power_function(5) == 25
+        assert await power_function(6) == 36
+
+    m_set.assert_called_once_with("6", 36, time_to_live=mock.ANY, if_not_exists=mock.ANY)
```
Comment (eigenein) [inline tests/decorators/test_async.py:151]: Picky/optional: it'd be easier to just check if the value was cached via `.get()`

**Review by eigenein — CHANGES_REQUESTED** [review]
The implementation looks good, I'd just prefer the simpler `test_exclude`

File: tests/decorators/test_async.py (hunk 1 above)
Comment (JeroennC) [reply tests/decorators/test_async.py:131]: Agreed, I thought it would still add a bit of value to testing the API explicitly but i don't mind leaving it out

File: tests/decorators/test_async.py (hunk 2 above)
Comment (JeroennC) [reply tests/decorators/test_async.py]: yess the less mocks the better :)

**Review by eigenein — APPROVED** [review]
